Default constructors for network devices and packets.

T0YMAK3R opened the pull request:
Set basic constructors for network devices and for packets.

T0YMAK3R committed:
Default constructors for network devices and packets.

nnickodem commented on src/main/java/Objects/NetworkDevices/Firewall.java:
```diff
@@ -1,4 +1,20 @@
+
+    @Override
+    public void filterPackets() {
+        //TODO: No idea how to implement this
```
wow such confidence

nnickodem commented on src/main/java/Objects/NetworkDevices/NetworkDevice.java:
```diff
@@ -10,7 +11,15 @@
+    private List<String> packets;
 
+    /**
+     *
```
you didnt remove my example java doc retard

T0YMAK3R replied:
It's because I'm still using it, you retard.

nnickodem commented on src/main/java/Objects/Packets/SYN.java:
```diff
@@ -2,9 +2,11 @@
+        setCost(1);
+        setStealth(1);
     }
 
```
what is this space?

T0YMAK3R replied:
1/10th of the room you need for your thumb

nnickodem commented on src/main/java/Objects/Packets/Crypto.java:
```diff
@@ -2,9 +2,11 @@
+        setCost(13);
+        setStealth(0);
     }
 
```
this one too!

T0YMAK3R replied:
With ten of these, you could fit your thumb in there.

nnickodem commented on src/main/java/Objects/Packets/Botnet.java:
```diff
@@ -2,10 +2,11 @@
+        setCost(1);
+        setStealth(0);
     }
 
```
this is chaos

T0YMAK3R replied:
So is walking around with those freakish weights on your hands.

nnickodem commented on src/main/java/Objects/NetworkDevices/Switch.java:
```diff
@@ -1,7 +1,19 @@
+        setId(id);
+    }
     //TODO: Switch specific
 
```
mass hysteria

T0YMAK3R replied:
Riot shields.

nnickodem commented on src/main/java/Objects/NetworkDevices/Router.java:
```diff
@@ -1,8 +1,30 @@
+
     //TODO: router specific attributes
     //route table? interface ips?
 
```
are you out of your mind with these spaces?

T0YMAK3R replied:
I'll add an extra tab next time.

nnickodem commented on src/main/java/Objects/NetworkDevices/NetworkDevice.java:
```diff
@@ -19,13 +28,49 @@ public String getId() {
+        return connections;
+    }
     //TODO: other universal values?
 
```
my god

T0YMAK3R replied:
Calm down, Martha Stewart.  Once it's compiled, it won't matter.  This isn't Python.

T0YMAK3R committed:
Updated comments.

nnickodem commented on src/main/java/Objects/NetworkDevices/NetworkDevice.java:
```diff
@@ -10,7 +11,15 @@
+    private int speed;
+    private Boolean hidden;
+    private int max_packet;
+    private List<String> packets;
```
no get speed/get packets function?

nnickodem commented on src/main/java/Objects/NetworkDevices/NetworkDevice.java:
```diff
@@ -19,13 +28,92 @@ public String getId() {
         return id;
```
this is the only one without javadoc?

T0YMAK3R replied:
Fixed.  Created javadoc for function.

nnickodem commented on src/main/java/Objects/NetworkDevices/NetworkDevice.java:
```diff
@@ -19,13 +28,92 @@ public String getId() {
+     * Gets whether or not the device is hidden by fog of war.
+     * @return Hidden variable.
+     */
+    public Boolean isHidden() {
```
typically you would write this as getHidden(). Letting intellij generate all these is the easiest way

T0YMAK3R replied:
When returning a boolean value, naming convention is the way I have it, not "get."
https://stackoverflow.com/questions/1370840/naming-conventions-what-to-name-a-method-that-returns-a-boolean

nnickodem commented on src/main/java/Objects/NetworkDevices/NetworkDevice.java:
```diff
@@ -19,13 +28,92 @@ public String getId() {
+     * Sets the types of packets each device can send.
+     * @param packet_type The types of packets each device can send.
+     */
+    public void setPacket(final String packet_type) {
```
if you are only setting one packet, wouldn't this be add packet? and when would we ever do that? Wouldn't a setPacket(final List<String> packets) suffice?

T0YMAK3R replied:
I felt this naming convention was better, because you'd use this when setting it (for example, you wouldn't take a device that can't send botnet packets and then add the ability for it to do so).
Doing the list way would include creating a list, instead of running a few add commands.  I don't see that is a better way of doing.

nnickodem replied:
So we would never be adding a packet to a device that it didn't already have correct? In that case we really should just be setting the list.

nnickodem replied:
unresolved, this is the same thing as setting connections, could be done easily using Arrays.asList

nnickodem commented on src/main/java/Objects/NetworkDevices/NetworkDevice.java:
```diff
@@ -1,16 +1,25 @@
     private ArrayList<Integer> connections;
+    private int speed;
+    private Boolean hidden;
+    private int max_packet;
```
not a huge deal but as I understand it camel-case is standard for java, so this would be maxPacket

T0YMAK3R replied:
Matter of style opinion.
Project for class.
Not a Java developer making a professional project.
Resolving.

nnickodem replied:
its to try and keep uniformity of code throughout the project so whoever wants to look at it can easily understand it

nnickodem commented on src/main/java/Objects/NetworkDevices/PC.java:
```diff
@@ -1,8 +1,40 @@
+     * @param max_packet    Max number of packets device can hold
+     * @param id            Unique device ID value
+     */
+    public PC(final int speed, final String team,
```
it looks like a lot of these constructors are setting the same variables, why not implement a constructor in NetworkDevice to set them?

T0YMAK3R replied:
Style issue.
Resolving.

nnickodem commented on src/main/java/Objects/NetworkDevices/Router.java:
```diff
@@ -1,8 +1,46 @@
+                    final Boolean hidden, final int max_packet,
+                    final String id) {
+
+        List<String> packets = new ArrayList();
```
never used

T0YMAK3R replied:
Removed unnecessary line.

nnickodem commented on src/main/java/Objects/Packets/Botnet.java:
```diff
@@ -1,11 +1,20 @@
+     * Constructor for the Botnet packet type.
+     * @param team Team that owns the packet.
+     */
+    public Botnet(final String team) {
```
Same as with NetworkDevice, why not a general packet constructor to handle team?

T0YMAK3R replied:
Good idea.
Fixing now.

nnickodem replied:
unchanged

nnickodem commented on src/main/java/Objects/Packets/Packet.java:
```diff
@@ -1,10 +1,78 @@
+    private String team;
+    private int cost;
+    private int stealth;
+    private String packet_type;
```
is this really needed? wont the class be enough?

T0YMAK3R replied:
Changed to "return getClass()"
Will need to figure out implementation for that for interaction with devices.

nnickodem replied:
We don't need the variable at all was the point

nnickodem replied:
The function that returns class is pointless as well

nnickodem commented on src/main/java/Objects/Packets/Packet.java:
```diff
@@ -1,10 +1,78 @@
+     * Sets the packet type.
+     * @param packet_type The type of packet it is.
+     */
+    public void setPacketType(final String packet_type) {
```
set-get set-get set-get get-set?

T0YMAK3R replied:
Right.
But it's style.  Can be fixed later.

nnickodem replied:
but why not fix now while you are working in it?

nnickodem commented on src/main/java/Objects/Packets/SYN.java:
```diff
@@ -1,10 +1,20 @@
+        setCost(1);
+        setStealth(1);
     }
 
```
this code breaks the program, particularity the fileHandler when it creates the objects. That should be amended even if it is just passing null variables

nnickodem replied:
program is still broken. Culprit appears to be your setPacket function not being a setPackets function that takes in a List

T0YMAK3R committed:
Made changes from comments.
Changed all ints to Integers.

nnickodem commented on src/main/java/Objects/NetworkDevices/NetworkDevice.java:
```diff
@@ -1,31 +1,131 @@
+     * Gets the last of packet types the device can send.
+     * @return The list of packet types the device can send.
+     */
+    public List getPacket() {
```
getPackets not getPacket

nnickodem commented on src/main/java/Objects/NetworkDevices/NetworkDevice.java:
```diff
@@ -1,31 +1,131 @@
+     * Gets the list of connected devices.
+     * @return List of connected devices.
+     */
+    public ArrayList<Integer> getConnections() {
```
`public List<Integer> getConnections()`

nnickodem commented on src/main/java/Objects/Packets/Packet.java:
```diff
@@ -1,10 +1,68 @@
+     * Gets what the packet type is (Botnet, Crypto, ICMP, SYN)
+     * @return The type of packet it is, as a class.
+     */
+    public Class getPacketType() {
```
this is pointless, you can just call the object's getClass()

T0YMAK3R committed:
Made Get-Set order consistent.
Changed setPacket to setPackets.
Made setPackets take a list as input instead of single strings.
Removed unnecessary function that returned class.

T0YMAK3R committed:
Some things were set as String in some places, and Integers in others.
Everything has been set to String.

T0YMAK3R committed:
Had to change more Integers to Strings.
